### ISSUE TITLE
Altering a shader?

### ISSUE THREAD

**Issue by peardox** [skia4delphi/skia4delphi#137]
Creating a shader includes these steps...
  Effect := TSkRuntimeEffect.MakeForShader(ShaderText, AErrorText);
  Painter := TSkPaint.Create;
  Painter.Shader := Effect.MakeShader(True);

Then assigning it to draw somehow - e.g.
  SkPaintBox1.Enabled := True;
  SkPaintBox1.OnDraw := SkPaintBox1Draw;

How to I reset the shader? I've tried re-creating, Reset set it to nil etc. but nothing seems to do anything.

Basically I'm trying to alter the shader by adding or removing a texture (the sksl is written to be aware of a missing texture)  

Sure I'm missing something dead simple (or do I have to re-create it completely?)

**Comment by peardox** [comment]
Thanks - that's the info I needed

**Comment by viniciusfbb** [comment]
> Tried it - perfect!

Nice!

> Oh - it is required to call if Effect.UniformExists('foo') before Effect.SetUniform('foo', bar)? I had a wierd bug in my stuff earlier when calling Effect.SetUniform() without first calling Effect.UniformExists()
> 
> I know that really I should always call Effect.UniformExists() but I knew that the 'foo' in question was present so skipped it (with annoying results it took me a while to track down)

No, it is not necessary to call Effect.UniformExists before Effect.SetUniform, as long as you know it exists.

If you can simulate this problem, and you can provide a project demonstrating this, I would appreciate it if you opened an issue for this.

**Comment by peardox** [comment]
> If you can simulate this problem, and you can provide a project demonstrating this, I would appreciate it if you opened an issue for this.

I wrote a testcase that had this issue earlier so I should be able to get back to it

If I didn't do something really stupid (quite possible) earlier I'll create a new issue
